### PR TITLE
Added windowin example

### PR DESCRIPTION
**Pull request by jaxpr**
This PR contains a windowing + RunInference example.
------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: sdks/python/apache_beam/examples/inference/milk_quality_prediction_windowing.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
+import argparse
```
Comment (damccorm) [inline sdks/python/apache_beam/examples/inference/milk_quality_prediction_windowing.py:1]: Please add a license header to this file

**Review by damccorm — COMMENTED** [review]
This mostly LGTM, could you please:

1) Add a brief description of how to run this/expected input and output to https://github.com/apache/beam/blob/master/sdks/python/apache_beam/examples/inference/README.md
2) Fix the linting violations flagged by CI
3) Add a writeup walking through the example on the site (like https://beam.apache.org/documentation/ml/multi-language-inference/)

File: sdks/python/apache_beam/examples/inference/milk_quality_prediction_windowing.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+  """
+  known_args, pipeline_args = parse_known_args(argv)
+  pipeline_options = PipelineOptions(pipeline_args)
+  pipeline_options.view_as(SetupOptions).save_main_session = save_main_session
+
+  df = pd.read_csv(known_args.input_data)
+  df['Grade'].replace(['low', 'medium', 'high'], [0, 1, 2], inplace=True)
+  x = df.drop(columns=['Grade'])
+  y = df['Grade']
+  x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.60, random_state=99)
```
[the remark this answers is not in the document]
Comment (jaxpr) [reply sdks/python/apache_beam/examples/inference/milk_quality_prediction_windowing.py]: I updated the code a bit so it should be more clear how the data is used. We use the training set to train the model and the test set as input data for the streaming pipeline.

**Comment by jaxpr** [comment]
@damccorm I have updated the PR and resolved your comments

File: sdks/python/apache_beam/examples/inference/README.md [hunk 3]
```diff
@@ -645,3 +645,44 @@ When all elements are in a single batch the output looks like this:
...
+    <INPUT_DATA> \
+    --training_set \
+    <TRAINING_SET> \
+    --labels \
+    <LABELS> \
+    --model_state \
+    <MODEL_STATE>
+```
+
+Where `<DATASET>` is the path to a csv file containing the Kaggle Milk Quality prediction dataset, `<INPUT_DATA>` the data that will be used as input for the streaming pipeline (test set), `<TRAINING_SET>` the path to store the training set in csv format, `<LABELS>` the path to store the csv containing the labels used to train the model and  `<MODEL_STATE>` the path to the JSON file containing the trained model.
```
Comment (damccorm) [inline sdks/python/apache_beam/examples/inference/README.md:677]:
```suggestion
Where `<DATASET>` is the path to a csv file containing the Kaggle Milk Quality prediction dataset, `<INPUT_DATA>` a filepath to save the data that will be used as input for the streaming pipeline (test set), `<TRAINING_SET>` a filepath to store the training set in csv format, `<LABELS>` a filepath to store the csv containing the labels used to train the model and  `<MODEL_STATE>` the path to the JSON file containing the trained model.
`<INPUT_DATA>`, `<TRAINING_SET>`, and `<LABELS>` will all be parsed from `<DATASET>` and saved before pipeline execution.
```

This was unclear to me just reading the description

File: sdks/python/apache_beam/examples/inference/README.md [hunk 4]
```diff
@@ -645,3 +645,44 @@ When all elements are in a single batch the output looks like this:
...
+
+Using the test set, we simulate a streaming pipeline that a receives a new measurement of the milk quality parameters every minute. A sliding window keeps track of the measurement of the last 30 minutes and new window starts every 5 minutes. The model predicts the quality of each measurement. After 30 minutes the results are aggregated in a tuple contains the number of measurements that were predicted as bad, medium and high quality samples. The output of each window looks as follows:
```
Comment (damccorm) [inline sdks/python/apache_beam/examples/inference/README.md:679]:
```suggestion
Using the test set, we simulate a streaming pipeline that a receives a new measurement of the milk quality parameters every minute. A sliding window keeps track of the measurement of the last 30 minutes and new window starts every 5 minutes. The model predicts the quality of each measurement. After 30 minutes the results are aggregated in a tuple containing the number of measurements that were predicted as bad, medium and high quality samples. The output of each window looks as follows:
```

File: sdks/python/apache_beam/examples/inference/milk_quality_prediction_windowing.py [hunk 5]
```diff
@@ -1,32 +1,78 @@
...
-      dest='input_data',
+      '--dataset',
+      dest='dataset',
+      required=True,
+      help='Path to the csv containing Kaggle Milk Quality dataset.')
+  parser.add_argument(
+      '--pipeline_input_data',
+      dest='pipeline_input_data',
+      required=True,
+      help='Path to store the csv containing input data for the pipeline.')
```
Comment (damccorm) [inline sdks/python/apache_beam/examples/inference/milk_quality_prediction_windowing.py:65]:
```suggestion
      help='Path to store the csv containing input data for the pipeline.'
      'This will be generated as part of preprocessing the data.')
```

Same nit as the README, this wasn't obvious to me from the comment alone.

File: sdks/python/apache_beam/examples/inference/milk_quality_prediction_windowing.py [hunk 6]
```diff
@@ -1,32 +1,78 @@
...
+  parser.add_argument(
+      '--training_set',
+      dest='training_set',
+      required=True,
+      help='Path to store the csv containing the training set')
```
Comment (damccorm) [inline sdks/python/apache_beam/examples/inference/milk_quality_prediction_windowing.py:70]:
```suggestion
      help='Path to store the csv containing the training set'
      'This will be generated as part of preprocessing the data.')
```

Same as other flag

File: sdks/python/apache_beam/examples/inference/milk_quality_prediction_windowing.py [hunk 7]
```diff
@@ -1,32 +1,78 @@
...
+  parser.add_argument(
+      '--labels',
+      dest='labels',
       required=True,
-      help='Path to the csv containing the dataset.')
+      help='Path to store the csv containing the labels used in training.')
```
Comment (damccorm) [inline sdks/python/apache_beam/examples/inference/milk_quality_prediction_windowing.py:75]:
```suggestion
      help='Path to store the csv containing the labels used in training.'
      'This will be generated as part of preprocessing the data.')
```

Same as other flag

**Review by damccorm — COMMENTED** [review]
Looks good, all comments are minor wording-level things

**Review by damccorm — APPROVED** [review]
LGTM - thanks!

File: sdks/python/apache_beam/examples/inference/milk_quality_prediction_windowing.py [hunk 8]
```diff
@@ -0,0 +1,240 @@
...
+      dest='training_set',
+      required=True,
+      help='Path to store the csv containing the training set.'
+           'This will be generated as part of preprocessing the data.')
+  parser.add_argument(
+      '--labels',
+      dest='labels',
+      required=True,
+      help='Path to store the csv containing the labels used in training.'
+           'This will be generated as part of preprocessing the data.')
```
Comment (damccorm) [inline sdks/python/apache_beam/examples/inference/milk_quality_prediction_windowing.py:79]:
```suggestion
      help='Path to store the csv containing input data for the pipeline.'
      'This will be generated as part of preprocessing the data.')
  parser.add_argument(
      '--training_set',
      dest='training_set',
      required=True,
      help='Path to store the csv containing the training set.'
      'This will be generated as part of preprocessing the data.')
  parser.add_argument(
      '--labels',
      dest='labels',
      required=True,
      help='Path to store the csv containing the labels used in training.'
      'This will be generated as part of preprocessing the data.')
```

Needs this to make the linter happy
